code/game/machinery/chemHandC.dm

```diff
@@ -0,0 +1,295 @@
+
+//Heater
+
+/obj/machinery/chemheater
+	name = "directed laser heater"
+	desc = "A platform with an integrated laser that uses high-energy photons to heat a subject through atomic vibrations. In a practical sense, it has no upper limit to how much thermal energy can be induced this way, as it is capable of reaching temperatures which could rapidly destroy any laboratory-approved container."
+	icon = 'icons/obj/chemHandC.dmi'
+	icon_state = "heater"
+	icon_state_open = "heater_open"
+	density = 1
+	anchored = 1
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | FIXED2WORK | EJECTNOTDEL
+	use_power = 1
+	idle_power_usage = 25
+	active_power_usage = 5000
+
+	var/max_temperature = TEMPERATURE_LASER
+	var/thermal_energy_transfer = 3000
+	var/laser_kind = 0
+	var/onstage = null
+
+	var/obj/item/weapon/reagent_containers/held_container
+	var/heating = FALSE
+	var/had_item = FALSE
+
+/obj/machinery/chemheater/New()
+	. = ..()
+
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/chemheater,
+		/obj/item/weapon/stock_parts/micro_laser,
+		/obj/item/weapon/stock_parts/capacitor
+	)
+	RefreshParts()
+	overlays += image(icon = icon, icon_state = "t[laser_kind]_laser") //Adds the side laser
```

Comment:
This isn't really necessary considering you do the exact same thing in refreshparts above